R/writeModel.R

```diff
@@ -25,62 +23,71 @@
 ##' removed before the BUGS code is saved.
 ##' In S-PLUS, a warning is generated when the model function is defined if the
 ##' last statement in the model is an assignment. To avoid this warning, add the
-##' line \code{invisible()} to the end of the model definition. This line will be
-##' removed before the BUGS code is saved.
+##' line \code{invisible()} to the end of the model definition. This line will
+##' be removed before the BUGS code is saved.
 ##'
-##' @param model R function containing the BUGS model in the BUGS
-##' model language, for minor differences see Section Details.
+
+#' @param model R function containing the JAGS model
+#'  in the BUGS ##' model language, for minor differences see Section Details.
 ##' @param con passed to \code{\link{writeLines}} which actually writes the
 ##' model file
 ##' @param digits number of significant digits used for \pkg{WinBUGS}
 ##' input, see \code{\link{formatC}}
-##' @return Nothing, but as a side effect, the model file is written
 ##'
 ##' @export
-##' @author original idea by Jouni Kerman, modified by Uwe Ligges, DSB removed S
-##' Plus part
-writeModel <- function(model, con = "model.bug", digits = 5)
-{
-    if (is.R()){
-        model.text <- c("model", replaceScientificNotationR(body(model),
-                                                            digits = digits))
-                                        # "[\+\-]?\d*\.?[Ee]?[\+\-]?\d*"
-    } else {
-        ## In S-PLUS the source code of a function can be obtained with
-        ## as.character(function_name).
-        ## This omits the "function_name <- function()" piece
-        model.text <- paste("model", as.character(model))
-    }
-    model.text <- gsub("%_%", "", model.text)
-    writeLines(model.text, con = con)
+##' @author original idea by Jouni Kerman, modified by Uwe Ligges,
+##'  DSB removed S-Plus part, WW modified the implementation.
+writeModel <- function(model, con = "model.bug", digits = 5) { # nolintr
```

Comment:
```suggestion
writeModel <- function(model, file = "model.bug", digits = 5) { # nolintr
```
maybe better argument name?